tests/__fixtures__/Rules/no_but_in_given_when/fixture.js

```diff
@@ -0,0 +1,105 @@
+const { format } = require("util");
+const generator = require("../../../helpers/problemGenerator");
+const NoButInGivenWhen = require("../../../../lib/rules/no_but_in_given_when");
+
+function generateProblem(location, keyword) {
+    return generator(
+        NoButInGivenWhen,
+        location,
+        format(NoButInGivenWhen.meta.message, keyword)
+    );
+}
+
+function getValidTestData() {
+    return [
+        [
+            "with Rule: Scenario",
+            `Feature: a feature file
+  Rule: a rule
+    Scenario: a scenario
+      Given a step
+      When a step
+      Then a step`,
+            [],
+        ],
+        [
+            "with Rule: Background",
+            `Feature: a feature file
+  Rule: a rule
+    Background: a background
+      Given a step
+    Scenario: a scenario
+      When a step
+      And a step
+      Then a step
+      And a step
+      But a step
+    Scenario Outline: a scenario outline
+      When a step
+      Then a step`,
+            [],
+        ],
+        [
+            "with Rule: Scenario Outline",
+            `Feature: a feature file
+  Rule: a rule
+    Scenario Outline: a scenario
+      Given a step
+      When a step
+      Then a step
+      But a step`,
+            [],
+        ],
+    ];
+}
+
+function getInvalidTestData() {
+    return [
+        [
+            "with Rule: Scenario",
+            `Feature: a feature file
+  Rule: a rule
+    Scenario: a scenario
+      Given a step
+      But a step
+      When a step
+      But a step
+      Then a step
+      But a step`,
+            [
+                generateProblem({ line: 5, column: 7 }, "Given"),
+                generateProblem({ line: 7, column: 7 }, "When"),
+            ],
+        ],
+        [
+            "with Rule: Background",
+            `Feature: a feature file
+  Rule: a rule
+    Background: a background
+      Given a step
+      And a step
```

Comment:
It would also be nice to have a case where there is `But` in the background `Given` section, to make sure that the code is processing this rule for the Background.

Reply:
Since the rule: `only_given_step_in_background` checks for But step in background,  `no_but_in_given_when` rule checks only for But steps in the Scenario/scenario outline but not in the background.
Should I make this rule to check in the Background also?

Reply:
If someone disables the `only_given_step_in_background` rule, and has the ` no_but_in_given_when` enabled, then they will expect to get a fail if there is a `But` step in any `Given` section.

So yes, can you please check the `Background` `Given` section in `no_but_in_given_when`

And I suppose that there might be multiple fail messages emitted for `But` in the Background section (it will often fail for 2 rules). But that should be OK.